src/bentoml/_internal/service/service.py

```diff
@@ -342,6 +353,18 @@ def openapi_spec(self) -> OpenAPISpecification:
 
         return generate_spec(self)
 
+    def on_startup(self, func: F_ctx) -> F_ctx:
+        self.startup_hooks.append(func)
```

Comment:
Would moving the context and `partial` call here clean things up a little maybe?

Reply:
Yeah, that would. In fact, that is the initial implementation: https://github.com/bentoml/BentoML/pull/3936/commits/b4afb466095b8c5290b2fc5e9f5bf3419afc34fd

But I want to keep the scope of context limited to where it is needed. If we move context to the service it will be created every time when a service is loaded. Though that wouldn't be a big deal I am okay for both. Let's see what others say.

Reply:
I think we can move the context and partial call here.